core/workspace_comment_render_svg.js

```diff
@@ -0,0 +1,465 @@
+      },
+      null);
+  var body = document.createElementNS(Blockly.HTML_NS, 'body');
+  body.setAttribute('xmlns', Blockly.HTML_NS);
+  body.className = 'blocklyMinimalBody';
+  var textarea = document.createElementNS(Blockly.HTML_NS, 'textarea');
+  textarea.className = 'blocklyCommentTextarea';
+  textarea.setAttribute('dir', this.RTL ? 'RTL' : 'LTR');
+  body.appendChild(textarea);
+  this.textarea_ = textarea;
+  this.foreignObject_.appendChild(body);
+  // Don't zoom with mousewheel.
+  Blockly.bindEventWithChecks_(textarea, 'wheel', this, function(e) {
+    e.stopPropagation();
+  });
+  Blockly.bindEventWithChecks_(textarea, 'change', this, function(
+      /* eslint-disable no-unused-vars */ e
+      /* eslint-enable no-unused-vars */) {
+    this.setContent(textarea.value);
+  });
+  return this.foreignObject_;
+};
+
+/**
+ * Add the resize icon to the DOM
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.prototype.addResizeDom_ = function() {
+  this.resizeGroup_ = Blockly.utils.createSvgElement(
+      'g',
+      {
+        'class': this.RTL ? 'blocklyResizeSW' : 'blocklyResizeSE'
+      },
+      this.svgGroup_);
+  var resizeSize = Blockly.WorkspaceCommentSvg.RESIZE_SIZE;
+  Blockly.utils.createSvgElement(
+      'polygon',
+      {'points': '0,x x,x x,0'.replace(/x/g, resizeSize.toString())},
+      this.resizeGroup_);
+  Blockly.utils.createSvgElement(
+      'line',
+      {
+        'class': 'blocklyResizeLine',
+        'x1': resizeSize / 3, 'y1': resizeSize - 1,
+        'x2': resizeSize - 1, 'y2': resizeSize / 3
+      }, this.resizeGroup_);
+  Blockly.utils.createSvgElement(
+      'line',
+      {
+        'class': 'blocklyResizeLine',
+        'x1': resizeSize * 2 / 3, 'y1': resizeSize - 1,
+        'x2': resizeSize - 1, 'y2': resizeSize * 2 / 3
+      }, this.resizeGroup_);
+};
+
+/**
+ * Add the delete icon to the DOM
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.prototype.addDeleteDom_ = function() {
+  this.deleteGroup_ = Blockly.utils.createSvgElement(
+      'g',
+      {
+        'class': 'blocklyCommentDeleteIcon'
+      },
+      this.svgGroup_);
+  this.deleteIconBorder_ = Blockly.utils.createSvgElement('circle',
+      {
+        'class': 'blocklyDeleteIconShape',
+        'r': '7',
+        'cx': '7.5',
+        'cy': '7.5'
+      },
+      this.deleteGroup_);
+  // x icon.
+  Blockly.utils.createSvgElement(
+      'line',
+      {
+        'x1': '5', 'y1': '10',
+        'x2': '10', 'y2': '5',
+        'stroke': '#fff',
+        'stroke-width': '2'
+      },
+      this.deleteGroup_);
+  Blockly.utils.createSvgElement(
+      'line',
+      {
+        'x1': '5', 'y1': '5',
+        'x2': '10', 'y2': '10',
+        'stroke': '#fff',
+        'stroke-width': '2'
+      },
+      this.deleteGroup_);
+};
+
+/**
+ * Handle a mouse-down on comment's resize corner.
+ * @param {!Event} e Mouse down event.
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.prototype.resizeMouseDown_ = function(e) {
```

Comment:
I noticed that resizing a comment doesn't seem to emit an event. I think this might be necessary for the vm to keep up-to-date information about the comment's position/shape (esp. for saving and loading a project).